src/test/java/ServerTest.java

```diff
@@ -1,39 +1,71 @@
 
 public class ServerTest {
 
     @Test
-    public void serverReceivesUsernameAndClientMessageAndPrintsToTerminal() throws IOException {
-        IOHelper stdIO = new IOHelper("");
-        IOHelper socketIO = new IOHelper("Devlin\nHello");
+    public void serverReceivesClientMessageAndPrintsToTerminal() throws IOException {
+        IOHelper stdIOclient1 = new IOHelper("");
+        IOHelper socketIOclient1 = new IOHelper("Hello");
 
-        Socket socketStub = new SocketStub(socketIO.getIn(), socketIO.getOut());
-        ServerSocket serverSocketStub = new ServerSocketStub(socketStub);
-        Server server = new Server(stdIO.getIn(), stdIO.getOut(), serverSocketStub);
+        Socket clientSocket1 = new SocketStub(socketIOclient1.getIn(), socketIOclient1.getOut());
+        ServerSocket serverSocketStub = new ServerSocketStub(Arrays.asList(clientSocket1));
+        new Server(stdIOclient1.getOut(), serverSocketStub);
 
-        server.start();
+        assertEquals(Message.clientConnected() + "\n" + "Hello\n", stdIOclient1.getOutput());
+    }
+
+    @Test
+    public void serverReceivesAndEchoesBackClientMessageToClientSocket() throws IOException {
+        IOHelper stdIOclient1 = new IOHelper("");
+        IOHelper socketIOclient1 = new IOHelper("Hello");
 
-        assertEquals(Message.clientConnected("Devlin") + "\n" +
-                Message.clientSays("Devlin") + "Hello\n", stdIO.getOutput());
+        Socket clientSocket1 = new SocketStub(socketIOclient1.getIn(), socketIOclient1.getOut());
+        ServerSocket serverSocketStub = new ServerSocketStub(Arrays.asList(clientSocket1));
+        new Server(stdIOclient1.getOut(), serverSocketStub);
+
+        assertEquals("Hello\n", socketIOclient1.getOutput());
     }
 
     @Test
-    public void serverReceivesAndEchoesBackClientMessageToClientSocketAfterUsername() throws IOException {
-        IOHelper stdIO = new IOHelper("");
-        IOHelper socketIO = new IOHelper("Devlin\nHello");
+    public void addClient() throws IOException {
+        IOHelper stdIOclient1 = new IOHelper("");
+        IOHelper socketIOclient1 = new IOHelper("");
+
+        IOHelper stdIOclient2 = new IOHelper("");
+        IOHelper socketIOclient2 = new IOHelper("");
+
+        Socket clientSocket1 = new SocketStub(socketIOclient2.getIn(), socketIOclient2.getOut());
+        ServerSocket serverSocketStub = new ServerSocketStub(Arrays.asList(clientSocket1));
+        Server server = new Server(stdIOclient2.getOut(), serverSocketStub);
 
-        Socket socketStub = new SocketStub(socketIO.getIn(), socketIO.getOut());
-        ServerSocket serverSocketStub = new ServerSocketStub(socketStub);
-        Server server = new Server(stdIO.getIn(), stdIO.getOut(), serverSocketStub);
 
-        server.start();
 
-        assertEquals(Message.clientSays("Devlin") + "Hello\n", socketIO.getOutput());
+        assertEquals("Hello\n", socketIOclient2.getOutput());
     }
+//    @Test
```

Comment:
Git is your friend if you want to remove something and find it again later 😄

Reply:
`git add -p` :)